compiler/fir/entrypoint/src/org/jetbrains/kotlin/fir/session/MetadataLibraryBasedSymbolProvider.kt

```diff
@@ -64,6 +67,22 @@ abstract class MetadataLibraryBasedSymbolProvider<L : MetadataLibrary>(
         }
     }
 
+    private fun getNameResolver(fragment: ProtoBuf.PackageFragment): NameResolver {
+        return fragmentToNameResolver.getOrPut(fragment.hashCode()) {
```

Comment:
I find this code questionable because it relies on the `hashCode` implementation in protobuf stubs which is highly dependent on the exact way how the stub was generated. And overall using hashCode as a key in hash map looks suspicious.

Can we use maybe `IdentityHasMap` here instead? Or some weak/soft map implementation?